core/templates/components/on-screen-keyboard/on-screen-keyboard.component.html

```diff
@@ -0,0 +1,239 @@
+<div class="oppia-osk-container" ng-if="$ctrl.showOSK()" ng-click="$ctrl.activateGuppy()">
+  <div class="oppia-osk-navbar">
+    <button ng-class="($ctrl.currentTab === 'mainTab') ? 'oppia-osk-btn-tab-active' : 'oppia-osk-btn-tab-inactive'" ng-click="$ctrl.changeTab('mainTab')">
+      123
+    </button>
+    <button ng-class="($ctrl.currentTab === 'functionsTab') ? 'oppia-osk-btn-tab-active' : 'oppia-osk-btn-tab-inactive'" ng-click="$ctrl.changeTab('functionsTab')">
+      f(x)
+    </button>
+    <button ng-class="($ctrl.currentTab === 'lettersTab') ? 'oppia-osk-btn-tab-active' : 'oppia-osk-btn-tab-inactive'" ng-click="$ctrl.changeTab('lettersTab')">
+      abc
+    </button>
+    <button class="oppia-osk-btn-hide" ng-click="$ctrl.hideOSK()">
+      HIDE <i class="fa fa-angle-down"></i>
+    </button>
+
+  </div>
+  <div class="oppia-osk-custom-letters-div" ng-if="$ctrl.customizable && $ctrl.currentTab === 'mainTab'">
+    <button class="oppia-osk-btn-letter" ng-click="$ctrl.insertString(letter)" ng-repeat="letter in $ctrl.customLetters"><[letter]></button>
+  </div>
+  <div class="oppia-osk-main-tab" ng-if="$ctrl.currentTab==='mainTab'">
+    <button class="oppia-osk-btn-number number-7" ng-click="$ctrl.insertString('7')">7</button>
```

Comment:
Yeah, I thought of that as well, but the thing is, this div contains elements of various types with different classes. There are buttons, images, and i tags. I don't think it'll be any easier to read and maintain if I were to turn this into a loop because I would have to add a lot of conditions and specific things, perhaps I would even have to add multiple loops, which is why I chose to implement it this way for this div and the one below it.

Reply:
Yeah, maybe not for numbers, but operators and functions seem doable (2 loops)? Since they have the same classes and similar 'click' functions.

Reply:
Done. I've used a loop for functions, but for operators, we only have 3 so we don't really need a loop, right?